### PR TITLE
fix(web): fix wrong type assertion caused error

### PR DESCRIPTION
**Pull request by irvingoujAtDevolution**
The value should be typed IronRdpError instead. We perhaps need to publish a small new patch for this.

DPS-10099

### CODE REVIEW

**Review by CBenoit — CHANGES_REQUESTED** [review]
I’m surprised by this. Isn’t that supposed to be the value returned on success? It is supposed to be `IronRdpError` in the `catchError`, and `SessionTerminationInfo` in the `map`…

**Comment by irvingoujAtDevolution** [comment]
> I’m surprised by this. Isn’t that supposed to be the value returned on success? It is supposed to be `IronRdpError` in the `catchError`, and `SessionTerminationInfo` in the `map`…

I think the logic goes like this:
``` rust
pub async fn run(&self) -> Result<SessionTerminationInfo, IronRdpError>
```
Rust async function returns a result, then wasm-bindgen maps it to `Promise<SessionTerminationInfo>` with the possibility to reject it with type `IronRdpError`

`from(session.run())` convert it to an observable, when returns/resolves to `SessionTerminationInfo`, it emits value like regular observable, but when reject/throw error, it emits error and caught by `catchError` operator. 

At the end, catchError returns `of(err)`, which passes down to map function.

You are right, after written this down, I realize it is possible for map to receive two values, it should be either `IronRdpError` or `SessionTerminationInfo`

**Comment by CBenoit** [comment]
@irvingoujAtDevolution Yes, I think the code was originally written assuming only the `Ok` value would be injected into the `map` closure. We need two distinct code paths: success or error, and the error should not be injected into the success code path. That would be the correct fix I think.

File: web-client/iron-remote-gui/src/services/wasm-bridge.service.ts [hunk 1]
```diff
@@ -191,7 +191,7 @@ export class WasmBridgeService {
                                 type: SessionEventType.TERMINATED,
                                 data: 'Session was terminated.',
                             });
-                            return of(err);
+                            throw err;
```
Comment (CBenoit) [inline web-client/iron-remote-gui/src/services/wasm-bridge.service.ts:194]: I read this code again, and I’m wondering if we need to re-throw the error at all.
```suggestion
```

Comment (irvingoujAtDevolution) [reply web-client/iron-remote-gui/src/services/wasm-bridge.service.ts:194]: we need, as the signature requires either to return an observable or break the control flow with throw.
`export declare function catchError<T, O extends ObservableInput<any>>(selector: (err: any, caught: Observable<T>) => O): OperatorFunction<T, T | ObservedValueOf<O>>;`

Comment (irvingoujAtDevolution) [reply web-client/iron-remote-gui/src/services/wasm-bridge.service.ts:194]: alternative is to use `tap` operator instead.

Comment (irvingoujAtDevolution) [reply web-client/iron-remote-gui/src/services/wasm-bridge.service.ts:194]: we may return a observable with nothing, like the `EMPTY` observable constant. which is practically the same in this case but throw an error really align better with the semantics RxJS provides.

Comment (CBenoit) [reply web-client/iron-remote-gui/src/services/wasm-bridge.service.ts:194]: Sounds good

**Review by CBenoit — APPROVED** [review]
Thank you for fixing this!

**Comment by irvingoujAtDevolution** [comment]
@CBenoit Let me know when there's a new build, I can update the version in DVLS

**Pull request merged**